crates/solana-listener/src/component/signature_realtime_scanner.rs

```diff
@@ -20,16 +21,48 @@ use crate::component::log_processor::fetch_logs;
 use crate::component::signature_batch_scanner;
 use crate::{SolanaTransaction, TxStatus};
 
+/// Environment variable that expects a base58 encoded signature
+/// of the last processed signature we want to force.
+const FORCE_LAST_PROCESSED_SIGNATURE: &str = "FORCE_LAST_PROCESSED_SIGNATURE";
+
 #[tracing::instrument(skip_all, err, name = "realtime log ingestion")]
 pub(crate) async fn process_realtime_logs(
     config: crate::Config,
-    latest_processed_signature: Option<Signature>,
     rpc_client: Arc<RpcClient>,
     mut signature_sender: MessageSender,
+    state: impl SolanaListenerState,
 ) -> Result<(), eyre::Error> {
     let gateway_program_address = config.gateway_program_address;
     let gas_service_config_pda = config.gas_service_config_pda;
 
+    let force_last_processed_signature = std::env::var(FORCE_LAST_PROCESSED_SIGNATURE)?;
+    let latest_processed_signature = if force_last_processed_signature.is_empty() {
+        state.latest_processed_signature()
+    } else {
+        tracing::warn!(
+            "forcing last processed signature to {}",
+            force_last_processed_signature
+        );
+        Some(Signature::from_str(&force_last_processed_signature)?)
+    };
+
+    // Fetch missed batches
+    let latest_signature = signature_batch_scanner::fetch_batches_in_range(
+        &config,
+        Arc::clone(&rpc_client),
+        &signature_sender,
+        latest_processed_signature,
+        None,
+    )
+    .instrument(info_span!("fetching missed signatures"))
+    .in_current_span()
+    .await?;
+
+    if let Some(latest_signature) = latest_signature {
+        // Send the latest signature
+        state.set_latest_processed_signature(latest_signature)?;
+    }
```

Comment:
We can recover the latest signature as `fetch_batches_in_range` returns it :) @roberts-pumpurs

Reply:
I tried the wrappping on the sender using traits, but i considered it too complex and pervasive in other parts of the code.